### ISSUE TITLE
Replace pretty-quick by a different tool.

### ISSUE THREAD

**Issue by rioug** [openfoodfoundation/openfoodnetwork#11823]
### What we should change and why (this is tech debt)

As first noted here https://github.com/openfoodfoundation/openfoodnetwork/pull/11165#issuecomment-1627870317 `pretty-quick` is not maintained anymore.
So we should replace it by something else,  maybe https://github.com/lint-staged/lint-staged is was mentioned a few times here :  https://github.com/azz/pretty-quick/issues/164

In the mean time we should not upgrade `prettier`

### Context

This came up here https://github.com/openfoodfoundation/openfoodnetwork/pull/11165

**Comment by mkllnk** [comment]
I don't think that we should install pre-commit hooks by default. It should be up to the developer when to run linters. And we have CI as last line of defense. You can integrate with your IDE or you may only change the README file and don't need to run any validations.

**Comment by rioug** [comment]
The pre commit hook was added here : https://github.com/openfoodfoundation/openfoodnetwork/commit/e0b40222ddc9c268ab172ee7e393e3f38660d183 
But I agree it should be part of CI

**Comment by dacook** [comment]
Something on my list of things to do when I have time (retire?) was to create a smart hook that only runs relevant linters depending on what files are changed. Also specs (eg rspec for `*_spec.rb`, jest for `*_test.js`, etc). 

But that's just my personal preference; I agree that it should be user-configurable. Maybe that's what Husky is for? 
_looks up husky_
Hmm I'm not sure what Husky is for exactly..

**Comment by binarygit** [comment]
I opened this https://github.com/openfoodfoundation/openfoodnetwork/pull/11824 which replaced `pretty-quick` with `lint-staged` and, had configured `lint-staged` to only run the liner for `js, json` files which are staged.

**Comment by SerenModz21** [comment]
Hi! I came across thie repo in the pretty-quick pr (https://github.com/prettier/pretty-quick/pull/165) and would just like to let you know that @JounQin has taken over the project and has released pretty-quick v4, which adds support for prettier v3.

related comments:
- https://github.com/prettier/pretty-quick/issues/164#issuecomment-1895400862
  > I've just finished my promise.
  > 
  > `pretty-quick` v4 has just been released with `prettier` v3 support! Thanks @SukkaW for helping as always! ❤️

- https://github.com/prettier/pretty-quick/pull/165#issuecomment-1891463129
  > I'll take over @azz's place and help to maintain this package in few days. 🍻

**Comment by rioug** [comment]
This was done as part of : #13675
